File: Minimum Size Subarray Sum.js
var minSubArrayLen = function(target, nums) {
    var right = 0;
    var left = 0;
    var sum = 0;
    var minLen = Infinity;
    while(right <= nums.length){
        if(sum < target){
            sum+=nums[right];
            right++;
        }else {
            minLen = Math.min(minLen, right-left);
            sum -= nums[left];
            left++;
        }
        
    }
    return minLen === Infinity ? 0 : minLen;
};

File: Minimum Size Subarray Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int result = nums.length;
        int window_sum = 0;
        int left = 0;
        boolean has_sub_array = false;
        
        for(int right = 0; right < nums.length; right++){
            window_sum += nums[right];
            while(window_sum >= target){
                has_sub_array = true;
                result = Math.min(result, right - left + 1);
                window_sum -= nums[left];
                left++;
            }
        }
        
        
        return has_sub_array? result : 0;
    }
}
